src/renderer/components/dock/create-resource.store.ts

```diff
@@ -1,14 +1,95 @@
+import fs from "fs-extra";
+import path from "path";
+import os from "os";
 import { autobind } from "../../utils";
 import { DockTabStore } from "./dock-tab.store";
 import { dockStore, IDockTab, TabKind } from "./dock.store";
+import { GroupSelectOption, SelectOption} from "../select";
+
+export interface BadgedSelectOption extends SelectOption {
+  badge?: string;
+}
+
+export interface BadgedGroupSelectOption extends GroupSelectOption<BadgedSelectOption> {
+  badge?: string;
+}
+
 
 @autobind()
 export class CreateResourceStore extends DockTabStore<string> {
+
   constructor() {
     super({
       storageName: "create_resource"
     });
   }
+
+  get lensTemplatesFolder():string {
+    return path.resolve(__static, "../templates/create-resource");
+  }
+
+  get userTemplatesFolder():string {
+    return path.join(os.homedir(), ".k8slens", "templates");
+  }
+
+  async getLensTemplates() {
+    const templates:BadgedGroupSelectOption[] = [];
+
+    fs.readdirSync(this.lensTemplatesFolder).forEach((group) => {
```

Comment:
please don't mix sync and async function. `fse` or `fs-extra` has this function as async.

Reply:
I think it would be better if we used something like https://www.npmjs.com/package/walk